src/lucky_record/charms/uuid_extensions.cr

```diff
@@ -0,0 +1,52 @@
+struct UUID
+  module Lucky
+    alias ColumnType = UUID
+    include LuckyRecord::Type
+
+    def parse(value : UUID)
+      SuccessfulCast(UUID).new(value)
+    end
+
+    # form form #set_id_from_param
+    def parse(value : String)
+      SuccessfulCast(UUID).new(UUID.new(value))
```

Comment:
I think a spec that gives it a bad UUID would be good. Then this should catch any errors raised if the UUID is incorrect and return a `FailedCast`